src/sql/workbench/services/notebook/common/sql/sqlSessionManager.ts

```diff
@@ -183,6 +186,29 @@ class SqlKernel extends Disposable implements nb.IKernel {
 		}
 	}
 
+	private setConnectionPath(): void {
+		if (this._path) {
+			let prefix = getUriPrefix(this._path);
+			if (!prefix || prefix === uriPrefixes.connection) {
+				this._connectionPath = uriPrefixes.notebook.concat(this._path);
+			} else if (prefix !== uriPrefixes.notebook) {
+				try {
+					let uri = URI.parse(this._path);
+					if (uri && uri.scheme) {
+						this._connectionPath = uri.toString().replace(uri.scheme, uriPrefixes.notebook);
+					}
+				} catch {
+					// Ignore exceptions from URI parsing
+				} finally {
+					// If _connectionPath hasn't been set yet, set _connectionPath to _path as a last resort
+					if (!this._connectionPath) {
+						this._connectionPath = this._path;
```

Comment:
You could just set this value initially and then it'll just be overwritten if that code runs